frontend/src/components/Notification/Notification.js

```diff
@@ -0,0 +1,146 @@
+import {
+  Grid,
+  Snackbar,
+  SnackbarContent,
+  Typography,
+  Button,
+  Tooltip,
+  IconButton,
+} from "@mui/material";
+import React, { useContext, useEffect, useState } from "react";
+import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
+import CheckCircleOutlineOutlinedIcon from "@mui/icons-material/CheckCircleOutlineOutlined";
+import EventSource from "eventsource";
+import { UserInfoContext } from "../../stores/UserInfoStore";
+
+const Notification = ({ addNotification, deleteNotification }) => {
+  const [showSnackbar, setShowSnackbar] = useState(false);
+  const [userInfo] = useContext(UserInfoContext);
+  const [notification, setNotification] = useState({
+    id: "",
+    title: "",
+    data: {},
+    content: "",
+  });
+  const [eventSource, setEventSource] = useState(undefined);
+
+  useEffect(() => {
+    if (!userInfo.loggedIn) return;
+    setEventSource(
+      new EventSource(
+        "http://localhost:8080/notification/sse?userId=" + userInfo.id,
+        {
+          headers: { Authorization: userInfo.jwt },
+        }
+      )
+    );
+  }, [userInfo]);
+
+  useEffect(() => {
+    if (eventSource === undefined) return;
```

Comment:
Would it be useful to do !!eventSource pour tout vérifié or just undefined is fine?

Reply:
Je l'initialise à undefined pour être sûr au début